pkg/operation/botanist/secrets.go

```diff
@@ -81,8 +82,18 @@ func (b *Botanist) lastSecretRotationStartTimes() map[string]time.Time {
 			}
 		}
 
-		if shootStatus.Credentials.Rotation.Kubeconfig != nil && shootStatus.Credentials.Rotation.Kubeconfig.LastInitiationTime != nil {
-			rotation[kubeapiserver.SecretStaticTokenName] = shootStatus.Credentials.Rotation.Kubeconfig.LastInitiationTime.Time
+		// Beginning with Kubernetes v1.27, the static token kubeconfig must always be disabled. Hence, there will not
+		// be any kubeconfig rotation anymore. However, the static token secret does also contain a token for the health
+		// check of the kube-apiserver which still must be rotated. Hence, let's use the last rotation initiation time
+		// of the CA rotation also to rotate the static token secret.
+		if versionutils.ConstraintK8sGreaterEqual127.Check(b.Shoot.KubernetesVersion) {
```

Comment:
Why don't we use the `EnableStaticTokenKubeconfig` flag instead of Kubernetes version here?
Clusters with Kubernetes `<1.27`  with static token kubeconfig disabled face the same problem, shouldn't they?
The `kubeconfig` rotation is not displayed on the dashboard anymore and even when it is started via annotations `.status.credentials.rotation.kubeconfig` of the shoot is gone as soon as the rotation is finished.

Reply:
You're right, that's a good point. I've incorporated your suggestion :)